packages/kolibri-common/components/SyncSchedule/EditDeviceSyncSchedule.vue

```diff
@@ -292,18 +298,19 @@
       },
 
       cancelBtn() {
-        this.$router.push({ name: PageNames.ManageSyncSchedule });
+        this.$router.push({ name: SyncPageNames.MANAGE_SYNC_SCHEDULE });
       },
       fetchDevice() {
         NetworkLocationResource.fetchModel({ id: this.$route.params.deviceId }).then(device => {
           this.device = device;
+          this.baseurl = device.base_url;
           TaskResource.list({ queue: 'facility_task' }).then(tasks => {
             this.tasks = tasks.filter(
               task =>
                 task.extra_metadata.device_id === device.id &&
                 task.facility_id === this.$store.getters.activeFacilityId
             );
-            if (this.tasks.length != 0) {
+            if (this.tasks) {
```

Comment:
You do need to do at least `if (this.tasks && this.tasks.length)` as otherwise if tasks is an empty Array, this will still be `true`.